From 3f6c6f1dd93744e98add7bc3656c13e204b51435 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Florian=20M=C3=BCllner?= <fmuellner@gnome.org>
Date: Fri, 6 Jun 2014 14:53:33 +0200
Subject: [PATCH] constraints: Translate frame to client rect for legacy
 fullscreen check

Stupid apps fullscreen themselves by resizing the client window to
monitor size. A monitor-sized frame rect on the other hand is perfectly
normal on monitors without struts - stop force-fullscreening those
and catch the real baddies instead.

https://bugzilla.gnome.org/show_bug.cgi?id=730681
---
 src/core/constraints.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/core/constraints.c b/src/core/constraints.c
index 250de41..f9b5668 100644
--- a/src/core/constraints.c
+++ b/src/core/constraints.c
@@ -411,6 +411,9 @@ setup_constraint_info (ConstraintInfo      *info,
    * if there are no struts making the workarea smaller than
    * the monitor.
    */
+
+  meta_window_frame_rect_to_client_rect (window, new, new);
+
   if (meta_prefs_get_force_fullscreen() &&
       window->client_type != META_WINDOW_CLIENT_TYPE_WAYLAND &&
       !window->hide_titlebar_when_maximized &&
@@ -429,6 +432,8 @@ setup_constraint_info (ConstraintInfo      *info,
       meta_window_make_fullscreen_internal (window);
     }
 
+  meta_window_client_rect_to_frame_rect (window, new, new);
+
   /* Log all this information for debugging */
   meta_topic (META_DEBUG_GEOMETRY,
               "Setting up constraint info:\n"
-- 
2.0.0

